tests/pass/move-data-across-await-point.rs

```diff
@@ -0,0 +1,84 @@
+use std::future::Future;
+
+// This test:
+// - Compares addresses of non-Copy data before and after moving it
+// - Writes to the pointer after it has moved across the await point
+//
+// This is only meant to assert current behavior, not guarantee that this is
+// how it should work in the future. In fact, upcoming changes to rustc
+// *should* break these tests.
+// See: https://github.com/rust-lang/rust/issues/62958
+async fn data_moved_async() {
+    async fn helper(mut data: Vec<u8>, raw_pointer: *mut Vec<u8>) {
+        let raw_pointer2 = &mut data as *mut Vec<u8>;
+        // When the Vec is moved into the generator upvar, that copies the Vec's
+        // ptr+capacity+len that is on the stack. Because of that, the value of
+        // `raw_pointer2` is different than that of `raw_pointer`.
+        //
+        // When copy propagation is enabled for generator upvars, the pointer
+        // addresses should be equal, becouse there will no longer be a
+        // generator upvar; instead, the function's `data` argument will be
+        // used directly.
```

Comment:
The function's `data` argument is still different from where `raw_pointer` points to though -- it points to the caller's `x`. So they would still be inequal I think?

Reply:
oh, yeah I think you are right. `data` itself is stored in another place than `x` even though the underlying Vec in the heap is the same. I'll remove that comment